### PR TITLE
cleanup: display file-sizes

### LINKED ISSUE #38404: Calling cleanup should display filesizes of purged formulae
Just a small suggestion for a minor enhancement. =)

Pouring new brews will display the filesize of the installed formulae:
```bash
==> Upgrading imagemagick
==> Downloading https://homebrew.bintray.com/bottles/imagemagick-6.9.1-0.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring imagemagick-6.9.1-0.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/imagemagick/6.9.1-0: 1442 files,  22M
```
So why not do the same when invoking Homebrew's `cleanup` command? It might look like this:
```bash
Johns-MacBook-Pro:Desktop johngardner$ brew cleanup
Removing: /usr/local/Cellar/imagemagick/6.9.0-10... (1442 files, 22M)
Removing: /usr/local/Cellar/little-cms2/2.6... (16 files, 1.1M)
Removing: /Library/Caches/Homebrew/imagemagick-6.9.0-10.mavericks.bottle.tar.gz...
Removing: /Library/Caches/Homebrew/little-cms2-2.6.mavericks.bottle.1.tar.gz...
Removed: 1458 files, 23.1M total
```
Might be nice to see how much disk space was saved after a clean-up; some formulae can be pretty hefty in volume (here's looking at *you*, MongoDB).

### PR DESCRIPTION
**Pull request by DomT4**
Closes #38404

### CODE REVIEW

File: Library/Homebrew/cmd/cleanup.rb [hunk 1]
```diff
@@ -64,12 +64,13 @@ def cleanup_formula f
   end
 
   def cleanup_keg keg
+    size = `/usr/bin/du -sh #{keg} | /usr/bin/awk -F'[^0-9]*' '{print $2}'`.strip
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/cleanup.rb]: Would be nice to pull this logic into Ruby rather than using pipes. Maybe worth looking at how `info.rb` does it?

**Comment by mikemcquaid** [comment]
Can you paste some sample output? Ta!

**Comment by DomT4** [comment]
Steps:

1) 
```
brew fetch https://raw.githubusercontent.com/Homebrew/homebrew/564f4f80721a33d0bcc5ec42054fd4fa1637cd85/Library/Formula/dmd.rb && brew install https://raw.githubusercontent.com/Homebrew/homebrew/564f4f80721a33d0bcc5ec42054fd4fa1637cd85/Library/Formula/dmd.rb && brew upgrade
```
2) ` brew cleanup `
3) ` brew remove dmd && rm -f /usr/local/etc/dmd.conf.old && rm -f /usr/local/etc/dmd.conf `

Benchmarks
==

WITH CHANGES:

time = 00:44:48
time2 = 01:29:32
time3 = 01:10:38
time4 = 01:24:89
time5 = 00:58:65

NO CHANGES:

time = 01:19:63
time2 = 01:10:54
time3 = 01:04:08
time4 = 01:05:08
time5 = 01:26:32

**Comment by mikemcquaid** [comment]
Sorry, I mean just copy-paste the output so I can see what it looks like :wink:

**Comment by DomT4** [comment]
> Sorry, I mean just copy-paste the output so I can see what it looks like

Sorry, my comment above was supposed to be pasted before your comments but you were ninja-quick in looking over this PR :wink:. The benchmarks were per Tim's request earlier :smile_cat:.

Visual output (Excuse my love of green terminals) -

screen shot 2015-04-06 at 20 53 15

File: Library/Homebrew/cmd/cleanup.rb (hunk 1 above)
Comment (DomT4) [inline Library/Homebrew/cmd/cleanup.rb]: Agreed, massively. I just wasn't sure how to grab folder sizes from Ruby given folders are basically, well, nothingness in themselves, it's the contents that contain the size, and I wasn't overly familiar with how Ruby handles sizing the entire contents of folders. I'm not even sure what I just said made much sense, hopefully it did. I'll look at the ` info.rb ` logic :smile_cat:

**Comment by mikemcquaid** [comment]
Maybe use parens instead of a comma to separate the sizes?

File: Library/Homebrew/cmd/cleanup.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/cleanup.rb]: You can use `Pathname#abv`

**Comment by DomT4** [comment]
Sure, no problem with that. I wasn't sure quite what style we wanted. Install uses ` /usr/local/Cellar/dmd/2.067.0: 362 files,  46M `.

File: Library/Homebrew/cmd/cleanup.rb [hunk 2]
```diff
@@ -64,12 +64,13 @@ def cleanup_formula f
...
     if keg.linked?
       opoo "Skipping (old) #{keg} due to it being linked"
     elsif ARGV.dry_run?
-      puts "Would remove: #{keg}"
+      puts "Would remove: #{keg}, #{size}M"
     else
-      puts "Removing: #{keg}..."
+      puts "Removing: #{keg}, #{size}M"
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/cleanup.rb]: Maybe using parenthesis to be consistent with what we do in `brew info` and `brew list`.

Comment (DomT4) [inline Library/Homebrew/cmd/cleanup.rb]: There's a discussion about the style in the general thread. Wasn't sure whether we wanted the style from ` install ` or from ` info `. Looks like parenthesis is the popular winner so far.

**Comment by jacknagel** [comment]
You can call `keg.abv` rather than reimplement the size calculation.

**Comment by DomT4** [comment]
> You can call keg.abv rather than reimplement the size calculation.

Does ` .abv ` work for the stuff in ` HOMEBREW_CACHE ` as well, or is the custom logic for size calculation on cached stuff okay?

**Comment by jacknagel** [comment]
It works on any pathname object:

```
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> :git.f.cached_download.abv
=> "3.5M"
irb(main):002:0> HOMEBREW_CACHE.abv
=> "3 files, 4.8M"
irb(main):003:0> :git.f.prefix.abv
=> "1363 files,  32M"
```

**Comment by DomT4** [comment]
Nice!

Heh, I essentially cooked up two methods when we already had a perfectly usable, solid one. Gave me the chance to play with Ruby for a couple hours this morning at least :smile_cat:.

**Comment by DomT4** [comment]
New visual output. I put the `...` back on the cleanup messages. Not sure if we want to keep them with the extra size information though? Easy enough to re-remove if maintainers prefer them :scissors:.

screen shot 2015-04-07 at 00 52 52

**Comment by DomT4** [comment]
Pushed :+1:

**Comment by mikemcquaid** [comment]
Nice turnaround here @DomT4!

**Comment by mikemcquaid** [comment]
Hmm, I just saw `Removing: /opt/boxen/cache/homebrew/src-0.18.yosemite.bottle.1.tar.gz... ( 28K)` when testing. Any idea how/why the extra space has crept in before `28K`?

**Comment by xu-cheng** [comment]
> Hmm, I just saw Removing: /opt/boxen/cache/homebrew/src-0.18.yosemite.bottle.1.tar.gz... ( 28K) when testing. Any idea how/why the extra space has crept in before 28K?

This patch should fix it:
```diff
diff --git a/Library/Homebrew/extend/pathname.rb b/Library/Homebrew/extend/pathname.rb
index 49adae2..1e80e3e 100644
--- a/Library/Homebrew/extend/pathname.rb
+++ b/Library/Homebrew/extend/pathname.rb
@@ -402,7 +402,7 @@ class Pathname
     out = ""
     n = Utils.popen_read("find", expand_path.to_s, "-type", "f", "!", "-name", ".DS_Store").split("\n").size
     out << "#{n} files, " if n > 1
-    out << Utils.popen_read("/usr/bin/du", "-hs", expand_path.to_s).split("\t")[0]
+    out << Utils.popen_read("/usr/bin/du", "-hs", expand_path.to_s).split("\t")[0].strip
     out
   end

```

**Comment by DomT4** [comment]
It's a white space from the Brew code I believe. I noticed it yesterday but just presumed the offset was intentional for clarity or such. Apologies, heh.

**Comment by DomT4** [comment]
Ah, apologies. Didn't see the email regarding Xu's patch. Do you want that rolled into this PR after testing?

**Comment by mikemcquaid** [comment]
Yes, please!

**Comment by DomT4** [comment]
Looks like the ` .strip ` fixed the issue :+1:.

screen shot 2015-04-07 at 16 52 29

Pushing now.

**Comment by DomT4** [comment]
Pushed.

**Comment by mikemcquaid** [comment]
Thanks @DomT4 !

**Comment by DomT4** [comment]
Thanks for the help Jack, Mike and Xu! :heart:

**Pull request closed without merging**